File: java/ReverseInteger.java
class ReverseInteger {
  public int reverse(int x) {
      int result = 0;
      

      while(x != 0) {
          int remainder = x % 10;
          x = x / 10;

          if(Integer.MAX_VALUE / 10 < result || result == Integer.MAX_VALUE / 10 && remainder > 7) {
              return 0;
          }
          if(result < Integer.MIN_VALUE / 10 || result == Integer.MIN_VALUE / 10 && remainder < -8) {
              return 0;
          }
          result = result * 10 + remainder;
              
      }
      
      
      
      
      return result;
  }
}

File: javascript/reverse-integer.js
/**
 * @param {number} x
 * @return {number}
 */
 var reverse = function(x) {
    
    let res = 0;
    
    
    
    let remainder;

    while(x >= 1) {
        remainder = x % 10;
        res = res * 10 + remainder;
        x = x / 10;
        }
    
    console.log(res);
    
    return res;
    
    
    
};